### PR TITLE
feat: add support for swift cocoapods lock files

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
add support for swift cocoapods lock files

Trivy-db doesn't include advisories for `swift`.
But found packages can be used in SBOM.

Podfile.lock:
```
PODS:
  - KeychainAccess (4.2.1)

COCOAPODS: 1.11.2
```

<details><summary> json result </summary>
  
```json
{
  "SchemaVersion": 2,
  "ArtifactName": "Podfile.lock",
  "ArtifactType": "filesystem",
  "Metadata": {
    "ImageConfig": {
      "architecture": "",
      "created": "0001-01-01T00:00:00Z",
      "os": "",
      "rootfs": {
        "type": "",
        "diff_ids": null
      },
      "config": {}
    }
  },
  "Results": [
    {
      "Target": "Podfile.lock",
      "Class": "lang-pkgs",
      "Type": "cocoapods",
      "Packages": [
        {
          "ID": "KeychainAccess/4.2.1",
          "Name": "KeychainAccess",
          "Version": "4.2.1",
          "Layer": {}
        }
      ]
    }
  ]
}

```
  
</details>

<details><summary> CycloneDX result </summary>

```json
{
  "bomFormat": "CycloneDX",
  "specVersion": "1.4",
  "serialNumber": "urn:uuid:47b6dea8-4e99-498c-b934-1e09461e84d5",
  "version": 1,
  "metadata": {
    "timestamp": "2022-12-05T08:50:03+00:00",
    "tools": [
      {
        "vendor": "aquasecurity",
        "name": "trivy",
        "version": "0.35.0-15-gb12f31f4"
      }
    ],
    "component": {
      "bom-ref": "88a26c98-ed95-4266-ae31-f7c5887453f4",
      "type": "application",
      "name": "integration/testdata/fixtures/fs/cocoapods",
      "properties": [
        {
          "name": "aquasecurity:trivy:SchemaVersion",
          "value": "2"
        }
      ]
    }
  },
  "components": [
    {
      "bom-ref": "pkg:swift/KeychainAccess@4.2.1",
      "type": "library",
      "name": "KeychainAccess",
      "version": "4.2.1",
      "purl": "pkg:swift/KeychainAccess@4.2.1",
      "properties": [
        {
          "name": "aquasecurity:trivy:PkgID",
          "value": "KeychainAccess/4.2.1"
        },
        {
          "name": "aquasecurity:trivy:PkgType",
          "value": "cocoapods"
        }
      ]
    },
    {
      "bom-ref": "80623adb-41ce-4122-9f42-a840b576fdfb",
      "type": "application",
      "name": "Podfile.lock",
      "properties": [
        {
          "name": "aquasecurity:trivy:Type",
          "value": "cocoapods"
        },
        {
          "name": "aquasecurity:trivy:Class",
          "value": "lang-pkgs"
        }
      ]
    }
  ],
  "dependencies": [
    {
      "ref": "80623adb-41ce-4122-9f42-a840b576fdfb",
      "dependsOn": [
        "pkg:swift/KeychainAccess@4.2.1"
      ]
    },
    {
      "ref": "88a26c98-ed95-4266-ae31-f7c5887453f4",
      "dependsOn": [
        "80623adb-41ce-4122-9f42-a840b576fdfb"
      ]
    }
  ],
  "vulnerabilities": []
}

```

</details>

<details><summary> SPDX json result </summary>

```json
{
        "SPDXID": "SPDXRef-DOCUMENT",
        "creationInfo": {
                "created": "2022-12-05T08:52:11.746332574Z",
                "creators": [
                        "Tool: trivy",
                        "Organization: aquasecurity"
                ]
        },
        "dataLicense": "CC0-1.0",
        "documentDescribes": [
                "SPDXRef-Filesystem-7cf45f15b814f677"
        ],
        "documentNamespace": "http://aquasecurity.github.io/trivy/filesystem/integration/testdata/fixtures/fs/cocoapods-8d62f2a6-fdf9-47cb-8beb-a1f614f1a567",
        "name": "integration/testdata/fixtures/fs/cocoapods",
        "packages": [
                {
                        "SPDXID": "SPDXRef-Application-4bddee9eba9c9e8d",
                        "filesAnalyzed": false,
                        "name": "cocoapods",
                        "sourceInfo": "Podfile.lock"
                },
                {
                        "SPDXID": "SPDXRef-Filesystem-7cf45f15b814f677",
                        "attributionTexts": [
                                "SchemaVersion: 2"
                        ],
                        "filesAnalyzed": false,
                        "name": "integration/testdata/fixtures/fs/cocoapods"
                },
                {
                        "SPDXID": "SPDXRef-Package-20b7bb58365129cc",
                        "attributionTexts": [
                                "PkgID: KeychainAccess/4.2.1"
                        ],
                        "externalRefs": [
                                {
                                        "referenceCategory": "PACKAGE-MANAGER",
                                        "referenceLocator": "pkg:swift/KeychainAccess@4.2.1",
                                        "referenceType": "purl"
                                }
                        ],
                        "filesAnalyzed": false,
                        "licenseConcluded": "NONE",
                        "licenseDeclared": "NONE",
                        "name": "KeychainAccess",
                        "versionInfo": "4.2.1"
                }
        ],
        "relationships": [
                {
                        "relatedSpdxElement": "SPDXRef-Filesystem-7cf45f15b814f677",
                        "relationshipType": "DESCRIBES",
                        "spdxElementId": "SPDXRef-DOCUMENT"
                },
                {
                        "relatedSpdxElement": "SPDXRef-Application-4bddee9eba9c9e8d",
                        "relationshipType": "CONTAINS",
                        "spdxElementId": "SPDXRef-Filesystem-7cf45f15b814f677"
                },
                {
                        "relatedSpdxElement": "SPDXRef-Package-20b7bb58365129cc",
                        "relationshipType": "CONTAINS",
                        "spdxElementId": "SPDXRef-Application-4bddee9eba9c9e8d"
                }
        ],
        "spdxVersion": "SPDX-2.2"
}

```

</details>

## Issues
- [ ] https://github.com/aquasecurity/trivy/issues/2915

## Related PRs
- [x] aquasecurity/go-dep-parser/pull/138

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [x] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by knqyf263** [comment]
Still work in progress?

**Comment by DmitriyLewen** [comment]
I updated PR for `cocoapods`.
But there are errors with tests after merge https://github.com/aquasecurity/go-dep-parser/pull/152.

I asked contributor to create PR with fix this.
If PR is not created in the near future - I will create a PR for this errors.

After that we can merge this PR.

File: pkg/detector/library/driver.go [hunk 1]
```diff
@@ -54,6 +54,9 @@ func NewDriver(libType string) (Driver, error) {
 		// Only semver can be used for version ranges
 		// https://docs.conan.io/en/latest/versioning/version_ranges.html
 		comparer = compare.GenericComparer{}
+	case ftypes.Cocoapods:
+		ecosystem = "cocoapods" // TODO use const from Trivy-db after adding it
+		comparer = compare.GenericComparer{}
```
Comment (knqyf263) [inline pkg/detector/library/driver.go:59]: Trivy doesn't support CocodPods for vulnerability scanning. I wonder we should return unsupported error.

**Review by knqyf263 — COMMENTED** [review]
Please ensure it properly generates CycloneDX and SPDX.

File: pkg/detector/library/driver.go (hunk 1 above)
Comment (knqyf263) [reply pkg/detector/library/driver.go:59]: Or, we can explicitly mention like "CocoaPods is supported for SBOM, not for vulnerability scanning".

Comment (DmitriyLewen) [reply pkg/detector/library/driver.go]: Added warning about this.

**Comment by DmitriyLewen** [comment]
> Please ensure it properly generates CycloneDX and SPDX.

CyclneDX and SPDX are correct. There are examples for this in the description.

**Comment by knqyf263** [comment]
@DmitriyLewen Do we also need to update the doc?

**Comment by DmitriyLewen** [comment]
@knqyf263 added some changes in docs. Take a look, please

File: docs/docs/sbom/supported.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+## Packages that support vulnerability scanning
+- [os packages][os_packages]
+- [language packages][os_packages]
```
Comment (knqyf263) [inline docs/docs/sbom/supported.md]:
```suggestion
- [language packages][language_packages]
```

Comment (knqyf263) [reply docs/docs/sbom/supported.md]: I believe it is just a mistake. Fixed in https://github.com/aquasecurity/trivy/pull/2956/commits/c449e70edeab979239d538574dec9d6b3d0a929f

Comment (knqyf263) [reply docs/docs/sbom/supported.md]: I believe it is just a mistake. Fixed in https://github.com/aquasecurity/trivy/pull/2956/commits/c449e70edeab979239d538574dec9d6b3d0a929f

**Review by knqyf263 — APPROVED** [review]

File: docs/docs/sbom/supported.md (hunk 2 above)
Comment (DmitriyLewen) [reply docs/docs/sbom/supported.md]: yes, thanks for fixing this.

**Pull request merged**